test/v2_integration.py

```diff
@@ -1661,3 +1661,77 @@ def test_auth_deactivation():
     resp = client.deactivate_authorization(order.authorizations[0])
     if resp.body.status is not messages.STATUS_DEACTIVATED:
         raise Exception("unexpected authorization status")
+
+def get_ocsp_response_and_reason(cert_file, issuer_file, url):
+    """Returns the ocsp response bytes and revocation reason."""
+    ocsp_request = make_ocsp_req(cert_file, issuer_file)
+    responses = fetch_ocsp(ocsp_request, url)
+
+    # Verify all responses are the same
+    for resp in responses:
+        if resp != responses[0]:
+            raise(Exception("OCSP responses differed: %s vs %s" %(
+                base64.b64encode(responses[0]), base64.b64encode(resp))))
+    ocsp_response = responses[0]
+
+    ocsp_resp_file = os.path.join(tempdir, "ocsp.resp")
+    with open(ocsp_resp_file, "wb") as f:
+        f.write(ocsp_response)
+
+    cmd = "openssl ocsp -no_nonce -issuer %s -cert %s -respin %s" % (
+        issuer_file, cert_file, ocsp_resp_file)
+    output = subprocess.check_output(cmd, shell=True).decode()
+    m = re.search('Reason: (\w+)', output)
+    reason = m.group(1) if m is not None else ""
+    return bytearray(ocsp_response), reason
+
+ocsp_response_bytes = None
+ocsp_response_reason = ""
+@register_twenty_days_ago
+def ocsp_resigning_setup():
+    """Issue and then revoke a cert in the past.
+
+    Useful setup for test_ocsp_resigning, which needs to check that the
+    revocation reason is still correctly set after re-signing and old OCSP
+    response.
+    """
+    client = chisel2.make_client(None)
+    order = chisel2.auth_and_issue([random_domain()], client=client)
+
+    cert = OpenSSL.crypto.load_certificate(
+        OpenSSL.crypto.FILETYPE_PEM, order.fullchain_pem)
+    reset_akamai_purges()
+    # Revoke for reason 1: keyCompromise
+    client.revoke(josepy.ComparableX509(cert), 1)
+
+    cert_file = os.path.join(tempdir, "compromised.pem")
+    with open(cert_file, "w") as f:
+        f.write(OpenSSL.crypto.dump_certificate(
+            OpenSSL.crypto.FILETYPE_PEM, cert).decode())
+
+    global ocsp_response_bytes, ocsp_response_reason
```

Comment:
Since these are globals, let's either name them with long descriptive names that tie them to the test, or define a dictionary with a name tied to the test and have these two as fields in it.

Reply:
Done. I've also just uploaded a much larger cleanup PR which uses this same pattern for other data-passing setups.